### DISCUSSION TITLE
Do arithmetic operations on values in different rows

### DISCUSSION THREAD

**Discussion by Alex-Cheng** [ClickHouse/ClickHouse#34608] [Q&A] [answered]
Arithmetic operations are usually applied on same rows, e.g. [Column A] + [Column B] where values of [Column A] and [B] are from the same rows. 

However, we need to apply arithmetic operations on different rows, e.g. [Column A] of the first row subtracts [Column A] of the following row, as shown as the table as following:
|   A    |  Result   |
|  ---    |    -----   |
|    2     |      -1     |
|    3     |      -3     |
|    6     |      1       |
|    5     |      null  |  

I know Join may help address the question, however it is inefficient. How can I do it in an efficient way? 

Thank you .

**Comment by UnamedRus** [comment, marked as answer]
LagInFrame + Window function

```
SELECT A - prev
FROM
(
    SELECT
        A,
        leadInFrame(A) OVER () AS prev
    FROM
    (
        SELECT toNullable(arrayJoin([2, 3, 6, 5])) AS A
    )
)

Query id: fb4a6cf0-634e-4aff-ac58-a926b608004d

┌─minus(A, prev)─┐
│             -1 │
│             -3 │
│              1 │
│           ᴺᵁᴸᴸ │
└────────────────┘
```

**Reply by Alex-Cheng** [reply]
Thank you. But why I cannot find the function `leadInFrame` in documentation? 

![image](https://user-images.githubusercontent.com/1518453/155068880-18030614-cf2d-4511-815a-c641e05e9825.png)

**Reply by UnamedRus** [reply]
Because there is no docs besides pr https://github.com/ClickHouse/ClickHouse/pull/21895

**Reply by Alex-Cheng** [reply]
Thank you. Why there is no docs for it? Is it because the function is too new or they are not officially offered? Can I use them with confidence?

**Reply by UnamedRus** [reply]
> Why there is no docs for it? 

Because developers doesn't like to write docs.

> Is it because the function is too new or they are not officially offered?

It's new.

> Can I use them with confidence?

Yes.
